sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -473,4 +447,76 @@ public CosmosAsyncDatabase getDatabase(String id) {
+                .map(response -> BridgeInternal.createFeedResponse(
+                    ModelBridgeInternal.getCosmosDatabasePropertiesFromV2Results(response.getResults()),
+                    response.getResponseHeaders()));
+        });
+    }
+
+
+    private Mono<CosmosDatabaseResponse> createDatabaseIfNotExistsInternal(CosmosAsyncDatabase database,
+                                                                           ThroughputProperties throughputProperties, Context context) {
+        String spanName = "createDatabaseIfNotExists." + database.getId();
+        Context nestedContext = context.addData(TracerProvider.COSMOS_CALL_DEPTH, TracerProvider.COSMOS_CALL_DEPTH_VAL);
+        Mono<CosmosDatabaseResponse> responseMono = createDatabaseIfNotExistsInternal(database.readInternal(new CosmosDatabaseRequestOptions(), nestedContext), database, throughputProperties, nestedContext);
+        return tracerProvider.traceEnabledCosmosResponsePublisher(responseMono,
+            context,
+            spanName,
+            database.getId(),
+            this.serviceEndpoint);
+    }
+
+    private Mono<CosmosDatabaseResponse> createDatabaseIfNotExistsInternal(Mono<CosmosDatabaseResponse> responseMono, CosmosAsyncDatabase database, ThroughputProperties throughputProperties, Context context) {
+        return responseMono.onErrorResume(exception -> {
+            final Throwable unwrappedException = Exceptions.unwrap(exception);
+            if (unwrappedException instanceof CosmosException) {
+                final CosmosException cosmosException = (CosmosException) unwrappedException;
+                if (cosmosException.getStatusCode() == HttpConstants.StatusCodes.NOTFOUND) {
+                    CosmosDatabaseRequestOptions requestOptions = new CosmosDatabaseRequestOptions();
+                    if(throughputProperties != null) {
+                        ModelBridgeInternal.setThroughputProperties(requestOptions, throughputProperties);
+                    }
+
+                    if (context != null) {
+                        Database wrappedDatabase = new Database();
+                        wrappedDatabase.setId(database.getId());
+                        return createDatabaseInternal(wrappedDatabase,
+                            requestOptions, context);
+                    }
+
+                    return createDatabase(new CosmosDatabaseProperties(database.getId()),
+                        requestOptions);
+                }
+            }
+            return Mono.error(unwrappedException);
+        });
+    }
+
+
+    private Mono<CosmosDatabaseResponse> createDatabaseInternal(Database database, CosmosDatabaseRequestOptions options,
+                                                             Context context) {
+        String spanName = "createDatabase." + database.getId();
+        Mono<CosmosDatabaseResponse> responseMono = createDatabaseInternal(database, options);
```

Comment:
Would like to keep it for readability

Reply:
@samvaity - we want more readability in these scenarios, where we are doing string concatenation inside the APIs. 
This doesn't cause any perf hits, since this `spanName` will just reference the already created string.